### PR TITLE
Created remove student from class route.

### PR DESCRIPTION
**Pull request by moustf**
- Created deleteStudentFromClass controller.
- Created deleteStudent from a class query.
- Tested deleting a student from a class route.
- Removed nodemon.json file.
- Added --runInBand flag to package.json.
- Imported functions and exported them from index files.

### CODE REVIEW

**Review by Bas-Shiekh — DISMISSED** [review]

File: server/routes/class.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import { userAuth, studentAndTeacher } from '../middlewares';
-import { getAnnouncement } from '../controllers';
+import { userAuth, studentAndTeacher, teacherAuth } from '../middlewares';
+import { getAnnouncement, deleteStudentFromARequest } from '../controllers';
 
 const classRouter = require('express').Router();
 
 classRouter.get('/class/:classId/announcement', userAuth, studentAndTeacher, getAnnouncement);
+classRouter.delete('/class/:classId/student/delete', userAuth, teacherAuth, deleteStudentFromARequest);
```
Comment (Raghad-Mezied) [inline server/routes/class.ts:7]: Don't add the method at the end of the endpoint, remove `delete`

File: server/queries/class/deleteStudentQuery.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Op } from 'sequelize';
+import { ClassStudent } from '../../models';
+
+const deleteStudentQuery = async (classId: string, studentId: string) => {
```
Comment (Raghad-Mezied) [inline server/queries/class/deleteStudentQuery.ts:4]: Let it be more clear name, you are just removing a student from the class, and the same for your controller.

File: server/queries/class/deleteStudentQuery.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const std = await ClassStudent.findOne({
+    where: {
+      [Op.and]: {
+        student_id: studentId,
+        class_id: classId,
+      },
+    },
+  });
```
Comment (Raghad-Mezied) [inline server/queries/class/deleteStudentQuery.ts:12]: Why do you find before destroying?

File: server/controllers/class/deleteStudentFromClass.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    const studentDeleted = await deleteStudentQuery(classId, studentId);
+
+    if (!studentDeleted) {
+      throw new CustomError(404, 'The student does not exist in this class!');
+    }
+
+    res.json({ msg: 'The student deleted successfully!', data: studentDeleted });
+  } catch (err) {
+    if (err.name === 'ValidationError') {
+      next(new CustomError(400, 'Wrong data is inserted!'));
```
Comment (Raghad-Mezied) [inline server/controllers/class/deleteStudentFromClass.ts:23]: Why you are handling the validation error when you are not doing any validation?

File: server/controllers/class/deleteStudentFromClass.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    } else next(new CustomError(err.status, err.msg));
+  }
```
Comment (Raghad-Mezied) [inline server/controllers/class/deleteStudentFromClass.ts:25]: If there is and status and msg for your error, then no need to customize it! for your controller, you will not get customized errors, and then you don't have a status or message.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Bas-Shiekh — DISMISSED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Aaya-Elsharief — DISMISSED** [review]

**Pull request merged**